File: c/ordereval.c
#include<stdio.h>
//#include<string.h>
//#include<malloc.h>
//#include<assert.h>




int main()
{

  int x = 5;
  int y = 5;

  int z = x + (x=0);
  int t = (y=0) + y;

  printf("starting from x = 5\n");
  printf("x + (x=0) evaluates to: %d\n",z);
  printf("(x=0) + x evaluates to: %d\n",t);


  return 0;


}

File: python/ordereval.py
#import sys

def main():

    x = 5
    y = 5

    def f():
        x = 0
        return x

    def g():
        y = 0
        return y

    print("Starting with x = 5\n")

    z = f() + x
    t = y + g()

    print("(x=0) + x evaluates to",z)
    print("x + (x=0) evaluates to",t)




if __name__ == '__main__':
    main()

